src/Microsoft.Restier.Core/Api.Interfaces.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Restier.Core
+{
+    [CLSCompliant(false)]
+    public interface IApiConfigurator
+    {
+        void Configure(IServiceCollection services, Type apiType);
+    }
+
+    public interface IApiInitializer
+    {
+        void Initialize(ApiConfiguration configuration);
+    }
+
+    public interface IApiContextConfigurator
+    {
+        void Initialize(ApiContext context);
+
+        void Cleanup(ApiContext context);
+    }
+
+    [CLSCompliant(false)]
+    public interface IApiContextFactory
+    {
+        ApiContext CreateWithin(IServiceScope scope);
+    }
+
+    public sealed class ApiTypeInfo
```

Comment:
Since we could potentially have multiple api types, this service is introduced to discover related info.
I suppose later we could establish some caches around it, like property/method cache, instead of using reflection in every individual service.